imports/api/users.js

```diff
@@ -258,14 +258,10 @@ Meteor.methods({
     })
   },
   
-  'users.getUserByEmail' (email) {
-  
-    if (!Meteor.user().hasGreaterRole('professor')) throw new Meteor.Error('invalid-permissions', 'Invalid permissions')
-    
+  'users.getUserIdByEmail' (email) {   
     const user = Meteor.users.findOne({ 'emails.0.address': email })
```

Comment:
I think you should use the Accounts find by email method. 
The way you have it, anyone could write a script to generate email address and then find the corresponding userId, so this is way too dangerous.
I suggest that for now, we don't implement sharing of questions between non-profs, so have add a check that the user is at least a professor, or an instructor for any course.
It's good that it returns Id instead of user, since we don't need user.